subsys/net/ip/tcp2.c

```diff
@@ -1341,6 +1341,12 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 	case TCP_ESTABLISHED:
 		/* full-close */
 		if (th && FL(&fl, ==, (FIN | ACK), th_seq(th) == conn->ack)) {
+			if (net_tcp_seq_cmp(th_ack(th), conn->seq) > 0) {
+				uint32_t len_acked = th_ack(th) - conn->seq;
+
+				conn_seq(conn, + len_acked);
```

Comment:
Quite unexpected syntax, but I see it's like this in other places in the new TCP stack so ok.

Reply:
Yes, I agree. I just do not want to change it right now as this style is used in other places in this file too.